test/unit/sharding_key_test.lua

```diff
@@ -0,0 +1,87 @@
+local t = require('luatest')
+local sharding_key_module = require('crud.common.sharding_key')
+local utils = require('crud.common.utils')
+
+local g = t.group('sharding_key')
+
+g.before_each(function()
+    local sharding_key_format = {
+        {name = 'space_name', type = 'string', is_nullable = false},
+        {name = 'sharding_key', type = 'array', is_nullable = false}
+    }
+    -- Create a space _ddl_sharding_key with a tuple that
+    -- contains a space name and it's sharding key.
+    if type(box.cfg) ~= 'table' then
+        box.cfg{}
+    end
+    box.schema.space.create('_ddl_sharding_key', {
+        format = sharding_key_format,
+        if_not_exists = true
+    })
+    box.space._ddl_sharding_key:create_index('pk', {if_not_exists = true})
```

Comment:
Caution: if this space already exists, then it may contain some data you don't expect in your test. I see `upsert()` below, but it does not protect from existence of meta for other spaces. It is better to fail early if something went wrong with cleaning between tests.

Reply:
I do actually cleanup after each testcase, so these `if_not_exists` are useless.

```lua
--- a/test/unit/sharding_key_test.lua
+++ b/test/unit/sharding_key_test.lua
@@ -17,15 +17,13 @@ g.before_each(function()
     end
     box.schema.space.create('_ddl_sharding_key', {
         format = sharding_key_format,
-        if_not_exists = true
     })
-    box.space._ddl_sharding_key:create_index('pk', {if_not_exists = true})
+    box.space._ddl_sharding_key:create_index('pk')
 end)
 
 g.after_each(function()
     -- Cleanup.
     cache.drop_caches()
-    box.space._ddl_sharding_key:truncate()
     box.space._ddl_sharding_key:drop()
 end)
 
```

Reply:
Fixed.

```diff
--- a/test/unit/sharding_key_test.lua
+++ b/test/unit/sharding_key_test.lua
@@ -18,14 +18,13 @@ g.before_each(function()
     box.schema.space.create('_ddl_sharding_key', {
         format = sharding_key_format,
     })
-    box.space._ddl_sharding_key:create_index('pk', {if_not_exists = true})
+    box.space._ddl_sharding_key:create_index('pk')
     box.schema.space.create('fetch_on_storage')
 end)
 
 g.after_each(function()
     -- Cleanup.
     if box.space._ddl_sharding_key ~= nil then
-        box.space._ddl_sharding_key:truncate()
         box.space._ddl_sharding_key:drop()
     end
     box.space.fetch_on_storage:drop()
```